### PR TITLE
cron-send-aws-instance-health.py

### PR DESCRIPTION
**Pull request by drewandersonnz**

### CODE REVIEW

**Comment by drewandersonnz** [comment]
ok to test

**Comment by drewandersonnz** [comment]
[test]

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+from openshift_tools.monitoring.metric_sender import MetricSender
+from openshift_tools.cloud.aws.base import Base
+
+def parse_args():
+    """ parse the args from the cli """
+    logger.debug("parse_args()")
+
+    parser = argparse.ArgumentParser(description='AWS instance health')
+    parser.add_argument('-v', '--verbose', action='store_true', default=None, help='Verbose?')
+    parser.add_argument('-vv', '--verbose2', action='store_true', default=None)
```
Comment (twiest) [inline scripts/monitoring/cron-send-aws-instance-health.py]: Consider `action='count'`

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    ms.add_metric({'openshift.cloud.aws.problems.SystemStatus': problems['SystemStatus']})
+    ms.add_metric({'openshift.cloud.aws.problems.Events': problems['Events']})
+
+    ms.send_metrics()
+    logger.info("Data sent to Zagg in %s seconds", str(time.time() - ms_time))
+
+def getRegions():
+    """ getRegions() """
+    logger.debug("getRegions()")
+    return Base.get_supported_regions()
```
Comment (twiest) [inline scripts/monitoring/cron-send-aws-instance-health.py]: Consider inlining this

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+#pylint: disable=import-error
+from openshift_tools.monitoring.metric_sender import MetricSender
+from openshift_tools.cloud.aws.base import Base
+
+def parse_args():
+    """ parse the args from the cli """
+    logger.debug("parse_args()")
+
+    parser = argparse.ArgumentParser(description='AWS instance health')
+    parser.add_argument('-v', '--verbose', action='store_true', default=None, help='Verbose?')
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-aws-instance-health.py]: change to count

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    """ send data to MetricSender"""
+    logger.debug("send_metrics(problems)")
+
+    ms_time = time.time()
+    ms = MetricSender()
+    logger.info("Send data to MetricSender")
+
+    ms.add_metric({'openshift.cloud.aws.problems.InstanceStatus': problems['InstanceStatus']})
+    ms.add_metric({'openshift.cloud.aws.problems.SystemStatus': problems['SystemStatus']})
+    ms.add_metric({'openshift.cloud.aws.problems.Events': problems['Events']})
```
Comment (twiest) [inline scripts/monitoring/cron-send-aws-instance-health.py]: Consider renaming these to:

`aws.ec2.instance.instance_status`
`aws.ec2.instance.system_status`
`aws.ec2.instance.events`

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+def send_metrics(problems):
+    """ send data to MetricSender"""
+    logger.debug("send_metrics(problems)")
+
+    ms_time = time.time()
+    ms = MetricSender()
+    logger.info("Send data to MetricSender")
+
+    ms.add_metric({'openshift.cloud.aws.problems.InstanceStatus': problems['InstanceStatus']})
```
Comment (drewandersonnz) [inline scripts/monitoring/cron-send-aws-instance-health.py]: rename to aws.ec2.instance.instance_status

File: scripts/monitoring/cron-send-aws-instance-health.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    logger.debug("regions: %s", regions)
+
+    problems = {
+        'CountAllInstances' : 0, # count all instances, including without problems
+        'InstanceStatus': 0,
+        'SystemStatus': 0,
+        'Events' : 0,
+    }
+
+    for region in regions:
```
Comment (twiest) [inline scripts/monitoring/cron-send-aws-instance-health.py:107]: Consider moving this section into its own function.

**Review by twiest — DISMISSED** [review]

File: scripts/monitoring/cron-send-aws-instance-health.py (hunk 6 above)
Comment (drewandersonnz) [reply scripts/monitoring/cron-send-aws-instance-health.py:107]: moved :-)

**Comment by drewandersonnz** [comment]
pylinted and tested on bastion. bot is missing pylint. email thread already started.

**Comment by BlueShells** [comment]
:+1:

**Pull request merged**